src/content/app/genome-browser/state/track-panel/trackPanelSlice.ts

```diff
@@ -14,43 +14,93 @@
 import { getActiveGenomePreviouslyViewedObjects } from './trackPanelSelectors';
 import { getActiveTrackPanel } from './trackPanelSelectors';
-import {
-  pickPersistentTrackPanelProperties,
-  TrackPanelStateForGenome
-} from './trackPanelState';
 
+import { ParsedUrlPayload } from 'src/content/app/genome-browser/state/browser-entity/browserEntitySlice';
+import { TrackSet } from 'src/content/app/genome-browser/components/track-panel/trackPanelConfig';
 import { RootState } from 'src/store';
-import { TrackSet } from '../trackPanelConfig';
 
-export const updateTrackPanelForGenome = createAction(
-  'track-panel/update-track-panel',
-  (payload: {
-    activeGenomeId: string;
-    data: Partial<TrackPanelStateForGenome>;
-  }) => {
-    const { activeGenomeId, data } = payload;
-    const persistentTrackProperties = pickPersistentTrackPanelProperties(data);
-    browserStorageService.updateTrackPanels({
-      [activeGenomeId]: persistentTrackProperties
-    });
-    return { activeGenomeId, data };
-  }
-)();
+export type PreviouslyViewedObject = {
+  genome_id: string;
+  object_id: string;
+  type: string;
+  label: string | string[];
+};
+
+export type PreviouslyViewedObjects = {
+  [genomeId: string]: PreviouslyViewedObject[];
+};
+
+export type TrackPanelStateForGenome = Readonly<{
+  isTrackPanelModalOpened: boolean;
+  isTrackPanelOpened: boolean;
+  selectedTrackPanelTab: TrackSet;
+  trackPanelModalView: string;
+  bookmarks: PreviouslyViewedObject[];
+  previouslyViewedObjects: PreviouslyViewedObject[];
+  highlightedTrackId: string;
+  collapsedTrackIds: string[];
+}>;
+
+export type TrackPanelState = Readonly<{
+  [genomeId: string]: TrackPanelStateForGenome;
+}>;
+
+export const defaultTrackPanelStateForGenome: TrackPanelStateForGenome = {
+  isTrackPanelModalOpened: false,
+  bookmarks: [],
+  previouslyViewedObjects: [],
+  selectedTrackPanelTab: TrackSet.GENOMIC,
+  trackPanelModalView: '',
+  highlightedTrackId: '',
+  isTrackPanelOpened: true,
+  collapsedTrackIds: []
+};
+
+export const getTrackPanelState = (): TrackPanelState => {
+  const genomeId = browserStorageService.getActiveGenomeId();
+  return genomeId ? { [genomeId]: getTrackPanelStateForGenome(genomeId) } : {};
+};
+
+export const getTrackPanelStateForGenome = (
+  genomeId: string
+): TrackPanelStateForGenome => {
+  return genomeId
+    ? {
+        ...defaultTrackPanelStateForGenome,
+        ...getPersistentTrackPanelStateForGenome(genomeId)
+      }
+    : defaultTrackPanelStateForGenome;
+};
+
+export const getPersistentTrackPanelStateForGenome = (
+  genomeId: string
+): Partial<TrackPanelStateForGenome> =>
+  browserStorageService.getTrackPanels()[genomeId] || {};
+
+export const pickPersistentTrackPanelProperties = (
+  trackPanel: Partial<TrackPanelStateForGenome>
+) => {
+  const persistentProperties = ['collapsedTrackIds', 'previouslyViewedObjects'];
+  return pick(trackPanel, persistentProperties);
+};
```

Comment:
Do you want to keep the selectors in the slice, or would you rather move them to their own file?

Reply:
I've moved them